feat: Add option to disable CloudProviders sidebar integration

eriolloan opened the pull request:
## Summary

Adds a new user setting to optionally disable CloudProviders D-Bus integration on Linux, allowing users to hide synchronized folders from the file manager sidebar while keeping sync functionality intact.

## Changes

- Adds checkbox in **Settings → General**: "Show synchronized folders in file manager sidebar"
- Default: **enabled** (maintains current behavior)
- When disabled: CloudProviders integration is not initialized
- **Linux-only** feature (CloudProviders is Linux-specific via libcloudproviders)

## Implementation details

- New config setting: `showCloudProvidersInFileManager` (defaults to `true`)
- UI checkbox only visible on Linux builds (`#ifndef Q_OS_LINUX`)
- Early return in `CloudProviderManager` constructor when disabled
- Requires application restart to take effect (CloudProviders initialized at startup)

## Testing

- [x] Checkbox appears in General Settings (Linux only)
- [x] Default is enabled
- [x] Disabling + restart removes folders from Nautilus sidebar
- [x] Re-enabling + restart restores folders
- [x] Sync functionality works in both states

## Screenshot
<img width="1192" height="283" alt="image" src="https://github.com/user-attachments/assets/153bb7b3-a69d-42bd-bd5b-2e343c57cafa" />

## Related issues

Fixes #1982

eriolloan committed:
feat: Add option to disable CloudProviders sidebar integration

Adds a new setting in General preferences to control whether synchronized folders appear in the file manager sidebar via CloudProviders on Linux.

Changes:
- Add 'showCloudProvidersInFileManager' config setting (defaults to true)
- Add checkbox in General Settings UI (Linux only)
- Check setting during CloudProviderManager initialization
- Skip DBus registration when disabled

The setting requires application restart to take effect.

Signed-off-by: eriolloan <15250783+eriolloan@users.noreply.github.com>

eriolloan committed:
refactor: Move CloudProviders check to Application initialization

Moves the setting check from CloudProviderManager constructor to
Application startup, following the suggestion to check before calling
setupCloudProviders rather than inside the manager itself.

This keeps the initialization logic cleaner and more consistent with
how other optional components are handled.

Signed-off-by: eriolloan <15250783+eriolloan@users.noreply.github.com>

eriolloan commented:
Thanks for the review! I've moved the check to application.cpp.

Regarding making the toggle take effect without restart - I looked into this and ran into a few complications. Tell me if I'm mistaken but it looks like :

1. `setupCloudProviders()` creates the manager with `new CloudProviderManager(this)` but doesn't store the pointer anywhere
2. To dynamically enable/disable, we'd need to store `_cloudProviderManager` in ownCloudGui and implement a cleanup method
3. More importantly, the manager registers with DBus (`g_bus_own_name`) during construction, and I'm not sure if simply destroying the object would properly unregister from DBus or if we'd need explicit cleanup

Is keeping the restart requirement acceptable for now? The tooltip does mention it's needed.
But i can look into what modifications would be needed.

camilasan commented:
> Thanks for the review! I've moved the check to application.cpp.

Thanks for the PR.

We discussed your changes in the team and I am afraid I didn't realize the implications of your changes when I first reviewed it.

So we would like to propose the following:
- remove the UI option: we think this new setting is very niche, most people will still use the default to enable the cloud providers. Keep it only in the config file.
  - for this I would suggest to also update the documentation so people will know it exists.

> 
> Regarding making the toggle take effect without restart - I looked into this and ran into a few complications. Tell me if I'm mistaken but it looks like :
> 
> 1. `setupCloudProviders()` creates the manager with `new CloudProviderManager(this)` but doesn't store the pointer anywhere
> 

I think you can leave this as it is now.

> 2. To dynamically enable/disable, we'd need to store `_cloudProviderManager` in ownCloudGui and implement a cleanup method
> 
> 3. More importantly, the manager registers with DBus (`g_bus_own_name`) during construction, and I'm not sure if simply destroying the object would properly unregister from DBus or if we'd need explicit cleanup
> 

Given that now it would require to change the config file by hand, then items 2 and 3 are not a concern anymore.

I hope this proposal is OK with you :)

mgallien reviewed (changes requested):
@eriolloan did you try bringing the point to the nautilus or other file managers using libcloudproviders to allow hiding the entries you do not want
I think you are trying to solve the issue at the wrong level

eriolloan committed:
refactor: Make CloudProviders setting config-only

Remove UI checkbox from General Settings.
Use showCloudProvidersInFileManager config file setting instead.

Signed-off-by: eriolloan <15250783+eriolloan@users.noreply.github.com>

JaniSoto commented:
Oh! This would be so useful! Not even editing a copy of the desktop file in .local/share/applications will do it.. After each update I have to manually edit the /usr/share/applications desktop file. It's pretty annoying because I have a bunch of synced folders and nautilus looks terrible... I really hope this gets merged!
